src/rpc/eth.rs

```diff
@@ -0,0 +1,957 @@
+        .unwrap())
+    }
+
+    async fn get_tx_by_block_number_and_index(
+        &self,
+        block_number: U64,
+        index: U64,
+    ) -> RpcResult<ethereum_jsonrpc::common::Tx> {
+        let txn = self.db.begin().await?;
+        let block_hash =
+            crate::accessors::chain::canonical_hash::read(&txn, BlockNumber(block_number.as_u64()))
+                .await?
+                .unwrap();
+        Ok(crate::accessors::chain::block_body::read_without_senders(
+            &txn,
+            block_hash,
+            BlockNumber(block_number.as_u64()),
+        )
+        .await?
+        .unwrap()
+        .transactions
+        .into_iter()
+        .nth(index.as_usize())
+        .map(|tx| ethereum_jsonrpc::common::Tx {
+            block_hash: Some(block_hash),
+            block_number: Some(block_number),
+            from: Address::zero(),
+            gas: U64::from(tx.message.gas_limit()),
+            gas_price: match tx.message {
+                Message::Legacy { gas_price, .. } => gas_price,
+                Message::EIP2930 { gas_price, .. } => gas_price,
+                Message::EIP1559 {
+                    max_fee_per_gas, ..
+                } => max_fee_per_gas,
+            },
+            hash: tx.message.hash(),
+            input: tx.message.input().clone(),
+            nonce: U64::from(tx.message.nonce()),
+            to: match tx.message.action() {
+                TransactionAction::Call(to) => Some(to),
+                _ => None,
+            },
+            transaction_index: Some(index),
+            value: tx.message.value(),
+            v: U64::from(tx.v()),
+            r: tx.r(),
+            s: tx.s(),
+        })
+        .unwrap())
+    }
+
+    async fn get_transaction_count(&self, address: Address, block_number: U64) -> RpcResult<U64> {
+        Ok(U64::from(
+            crate::accessors::state::account::read(
+                &self.db.begin().await?,
+                address,
+                Some(BlockNumber(block_number.as_u64())),
+            )
+            .await?
+            .unwrap()
+            .nonce,
+        ))
+    }
+
+    async fn get_transaction_receipt(
+        &self,
+        hash: H256,
+    ) -> RpcResult<ethereum_jsonrpc::common::TxReceipt> {
+        let txn = self.db.begin().await?;
+        let block_number = crate::accessors::chain::tl::read(&txn, hash)
+            .await?
+            .unwrap();
+        let block_hash = crate::accessors::chain::canonical_hash::read(&txn, block_number)
+            .await?
+            .unwrap();
+        let header = PartialHeader::from(
+            crate::accessors::chain::header::read(&txn, block_hash, block_number)
+                .await?
+                .unwrap(),
+        );
+        let block_body =
+            crate::accessors::chain::block_body::read_with_senders(&txn, block_hash, block_number)
+                .await?
+                .unwrap();
+        let mut state = InMemoryState::default();
```

Comment:
It does not make any sense, you need to re-execute block on top of the same state (block_number - 1) to get accurate receipts.